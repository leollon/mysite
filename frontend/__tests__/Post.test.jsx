// __tests__/Post.test.jsx

import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { create } from 'react-test-renderer';

import ArticleList from '../components/Post';

let container = null;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe('rendering Post component', () => {
    it('renders with empty articles', () => {
        const articles = [];

        render(<ArticleList articles={articles} />, container);
        expect(container.textContent).toBe('No Articles yet!');

        // snapshot testing
        const component = create(<ArticleList articles={articles} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders with more than one articles', () => {
        let articles = [
            {
                title: 'abc',
                category: 'ddd',
                author: 'root',
                created_time: '2019-11-16 06:11:19 +0800',
                comment_statistics: 0,
                article_body: 'text content',
                slug: 'abcdads',
                user_view_times: 0,
                tags: 'untagged',
            },
            {
                title: 'cap theory',
                category: 'theory',
                author: 'root',
                created_time: '2019-11-16 06:11:19 +0800',
                comment_statistics: 0,
                article_body: 'Consistency, Availability, Partition tolerance',
                slug: 'cap-theory',
                user_view_times: 0,
                tags: 'cap,分布式系统理论',
            },
            {
                title: 'code',
                category: 'python',
                author: 'root',
                created_time: '2019-11-16 06:11:19 +0800',
                comment_statistics: 0,
                article_body:
                    '```python\n\nprint("hello")\n\nprint("hello")\n\nprint("hello")\n\n```',
                slug: 'code',
                user_view_times: 0,
                tags: 'highlight,dynamic-import',
            },
            {
                title: 'test-head',
                category: 'uncategorized',
                author: 'root',
                created_time: '2019-11-16 06:11:19 +0800',
                comment_statistics: 0,
                article_body: '# head1',
                slug: 'test-head',
                user_view_times: 0,
                tags: 'tag1,tag2,tag3,tag4,tag5',
            },
        ];

        render(<ArticleList articles={articles} />, container);
        // title
        expect(container.textContent).toContain('abc');
        expect(container.textContent).toContain('cap theory');
        expect(container.textContent).toContain('code');
        expect(container.textContent).toContain('test-head');

        // category
        expect(container.textContent).toContain('uncategorized');
        expect(container.textContent).toContain('python');
        expect(container.textContent).toContain('theory');
        expect(container.textContent).toContain('ddd');

        // tag
        expect(container.textContent).toContain('#untagged');
        expect(container.textContent).toContain('#cap, #分布式系统理论');
        expect(container.textContent).toContain('#highlight, #dynamic-import');
        expect(container.textContent).toContain(
            '#tag1, #tag2, #tag3, #tag4, #tag5'
        );

        // snapshot testing
        const component = create(<ArticleList articles={articles} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
