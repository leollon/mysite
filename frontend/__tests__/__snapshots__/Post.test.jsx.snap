// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Post component renders with empty articles 1`] = `
<div
  className="post-preview"
>
  <h2
    className="post-title empty"
  >
    No Articles yet!
  </h2>
</div>
`;

exports[`rendering Post component renders with more than one articles 1`] = `
Array [
  <div
    className="post-preview"
  >
    <h2
      className="post-title"
    >
      <a
        href="/articles/abcdads"
      >
        abc
      </a>
    </h2>
    <article
      className="post-subtitle"
    >
      <p>
        text content....
      </p>
    </article>
    <div
      className="article-extra"
    >
      <span
        className="fa fa-folder"
      >
        <a
          href="/categories/ddd"
        >
          ddd
        </a>
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/untagged"
        >
          #
          untagged
        </a>
        
      </span>
      <span
        className="fa fa-comments"
      >
        <a
          href="/articles/abcdads#comments"
        >
          <span>
            0
          </span>
        </a>
      </span>
      <span
        className="fa fa-eye"
      >
        0
      </span>
    </div>
    <p
      className="post-meta"
    >
      Posted by 
      root
       on
       
      2019-11-16 06:11:19 +0800
    </p>
  </div>,
  <div
    className="post-preview"
  >
    <h2
      className="post-title"
    >
      <a
        href="/articles/cap-theory"
      >
        cap theory
      </a>
    </h2>
    <article
      className="post-subtitle"
    >
      <p>
        Consistency, Availability, Partition tolerance....
      </p>
    </article>
    <div
      className="article-extra"
    >
      <span
        className="fa fa-folder"
      >
        <a
          href="/categories/theory"
        >
          theory
        </a>
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/cap"
        >
          #
          cap
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/分布式系统理论"
        >
          #
          分布式系统理论
        </a>
        
      </span>
      <span
        className="fa fa-comments"
      >
        <a
          href="/articles/cap-theory#comments"
        >
          <span>
            0
          </span>
        </a>
      </span>
      <span
        className="fa fa-eye"
      >
        0
      </span>
    </div>
    <p
      className="post-meta"
    >
      Posted by 
      root
       on
       
      2019-11-16 06:11:19 +0800
    </p>
  </div>,
  <div
    className="post-preview"
  >
    <h2
      className="post-title"
    >
      <a
        href="/articles/code"
      >
        code
      </a>
    </h2>
    <article
      className="post-subtitle"
    >
      <pre>
        <code
          className="language-python"
        >
          print("hello")

print("hello")

print("hello")

\`\`\`....
        </code>
      </pre>
    </article>
    <div
      className="article-extra"
    >
      <span
        className="fa fa-folder"
      >
        <a
          href="/categories/python"
        >
          python
        </a>
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/highlight"
        >
          #
          highlight
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/dynamic-import"
        >
          #
          dynamic-import
        </a>
        
      </span>
      <span
        className="fa fa-comments"
      >
        <a
          href="/articles/code#comments"
        >
          <span>
            0
          </span>
        </a>
      </span>
      <span
        className="fa fa-eye"
      >
        0
      </span>
    </div>
    <p
      className="post-meta"
    >
      Posted by 
      root
       on
       
      2019-11-16 06:11:19 +0800
    </p>
  </div>,
  <div
    className="post-preview"
  >
    <h2
      className="post-title"
    >
      <a
        href="/articles/test-head"
      >
        test-head
      </a>
    </h2>
    <article
      className="post-subtitle"
    >
      <h1>
        head1....
      </h1>
    </article>
    <div
      className="article-extra"
    >
      <span
        className="fa fa-folder"
      >
        <a
          href="/categories/uncategorized"
        >
          uncategorized
        </a>
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/tag1"
        >
          #
          tag1
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/tag2"
        >
          #
          tag2
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/tag3"
        >
          #
          tag3
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/tag4"
        >
          #
          tag4
        </a>
        , 
      </span>
      <span
        className="fa fa-tag"
      >
        <a
          href="/tags/tag5"
        >
          #
          tag5
        </a>
        
      </span>
      <span
        className="fa fa-comments"
      >
        <a
          href="/articles/test-head#comments"
        >
          <span>
            0
          </span>
        </a>
      </span>
      <span
        className="fa fa-eye"
      >
        0
      </span>
    </div>
    <p
      className="post-meta"
    >
      Posted by 
      root
       on
       
      2019-11-16 06:11:19 +0800
    </p>
  </div>,
]
`;
